processing/src/main/java/io/druid/query/topn/PooledTopNAlgorithm.java

```diff
@@ -184,6 +203,7 @@ protected void scanAndAggregate(
 
     final int aggSize = theAggregators.length;
     final int aggExtra = aggSize % AGG_UNROLL_COUNT;
+    final AtomicInteger currentPosition = new AtomicInteger(0);
```

Comment:
a regular int value should be enough here, everything happens within the same thread